src/sinks/honeycomb/config.rs

```diff
@@ -0,0 +1,188 @@
+    #[serde(default)]
+    request: TowerRequestConfig,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    encoding: Transformer,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+fn default_endpoint() -> String {
+    "https://api.honeycomb.io/1/batch".to_string()
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+struct HoneycombDefaultBatchSettings;
+
+impl SinkBatchSettings for HoneycombDefaultBatchSettings {
+    const MAX_EVENTS: Option<usize> = None;
+    const MAX_BYTES: Option<usize> = Some(100_000);
+    const TIMEOUT_SECS: f64 = 1.0;
+}
+
+impl GenerateConfig for HoneycombConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"api_key = "${HONEYCOMB_API_KEY}"
+            dataset = "my-honeycomb-dataset""#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "honeycomb")]
+impl SinkConfig for HoneycombConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let batch_settings = self.batch.validate()?.into_batcher_settings()?;
+
+        let request_builder = HoneycombRequestBuilder {
+            encoder: HoneycombEncoder {
+                transformer: self.encoding.clone(),
+            },
+        };
+
+        let honeycomb_service_request_builder = HoneycombSvcRequestBuilder {
+            uri: self.build_uri(),
+            api_key: self.api_key.clone(),
+        };
+
+        let client = HttpClient::new(None, cx.proxy())?;
+
+        let service = HttpService::new(client.clone(), honeycomb_service_request_builder);
+
+        let request_limits = self.request.unwrap_with(&TowerRequestConfig::default());
+
+        let retry_logic =
+            HttpStatusRetryLogic::new(|req: &HttpResponse| req.http_response.status());
+
+        let service = ServiceBuilder::new()
+            .settings(request_limits, retry_logic)
+            .service(service);
+
+        let sink = HoneycombSink::new(service, batch_settings, request_builder);
+
+        let healthcheck = healthcheck(self.build_request(Vec::new())?, client).boxed();
+
+        Ok((VectorSink::from_event_streamsink(sink), healthcheck))
+    }
+
+    fn input(&self) -> Input {
+        let requirement = Requirement::empty().optional_meaning("timestamp", Kind::timestamp());
+
+        Input::log().with_schema_requirement(requirement)
+    }
+
+    fn acknowledgements(&self) -> &AcknowledgementsConfig {
+        &self.acknowledgements
+    }
+}
+
+impl HoneycombConfig {
+    fn build_uri(&self) -> Uri {
+        let uri = format!("{}/{}", self.endpoint, self.dataset);
+
+        uri.parse::<Uri>().expect("This should be a valid uri")
+    }
```

Comment:
Outside the scope of this PR, so feel free to ignore, but this function should ideally return a result instead of panicking..